src/api_manager/context/request_context.cc

```diff
@@ -447,17 +448,22 @@ std::string RequestContext::GetBackendPath() const {
     std::string parameters;
     for (std::size_t i = 0; i != method_call_.variable_bindings.size(); i++) {
       auto &variable_binding = method_call_.variable_bindings[i];
-      if (variable_binding.field_path.size() == 1) {
-        parameters += variable_binding.field_path[0];
-      } else if (variable_binding.field_path.size() > 1) {
-        parameters = std::accumulate(variable_binding.field_path.begin(),
-                                     variable_binding.field_path.end(),
-                                     std::string("."));
-      }
-      parameters.append("=");
-      parameters.append(variable_binding.value);
-      if (i != method_call_.variable_bindings.size() - 1) {
-        parameters.append("&");
+      for (std::size_t j = 0; j < variable_binding.field_path.size(); ++j) {
+        // If field_path is snake case, need to use corresponding jsonName.
+        std::string::size_type found = variable_binding.field_path[j].find("_");
+        std::string field_path;
+        if (found != std::string::npos &&
+            service_context_->config()->GetJsonName(
```

Comment:
How not not found from GetJsonName

Reply:
the origin snake case path will be used.

Reply:
Also add test cases.